test/gurka/test_maxspeed.cc

```diff
@@ -0,0 +1,30 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+TEST(Standalone, Maxspeed) {
+  const std::string ascii_map = R"(
+      A--B--C--D--E--F
+  )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "motorway"}, {"maxspeed", "50"}}},
+      {"BC", {{"highway", "motorway"}, {"maxspeed", "60mph"}}},
+      {"CD", {{"highway", "motorway"}, {"maxspeed", ""}}},
+      {"DE", {{"highway", "motorway"}}},
+      {"EF", {{"highway", "motorway"}, {"maxspeed", "none"}}},
+  };
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_maxspeed");
+  auto result = gurka::route(map, "A", "F", "auto");
+
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  auto leg = result.trip().routes(0).legs(0);
+
+  EXPECT_EQ(leg.node(0).edge().speed_limit(), 50);  // AB
+  EXPECT_EQ(leg.node(1).edge().speed_limit(), 97);  // BC
+  EXPECT_EQ(leg.node(2).edge().speed_limit(), 0);   // CD
+  EXPECT_EQ(leg.node(3).edge().speed_limit(), 0);   // DE
```

Comment:
Given that `speed_limit` is optional in `proto/trip.proto`, is checking for `0` the right thing here?  Shouldn't it be undefined?

Reply:
Not sure. Seems like trip_edge.speed_limit is getting set with a default value from edgeinfo [here](https://github.com/valhalla/valhalla/blob/master/src/thor/triplegbuilder.cc#L873).

Reply:
Hmm, how do we store "unlimited" (to support the Autobahns in .de?). Is the design here finalized?

Reply:
"unlimited" is tested in the next case `EF`. The trip_edge proto stores speed_limit 255, and we later serialize that out as `"speed_limit": "unlimited"`.

Reply:
Hmm, I wonder if this could be more explicitly expressed in the `.proto` using a [oneof](https://developers.google.com/protocol-buffers/docs/proto#oneof) expression:

```protobuf

message Edge {
   ...
  enum SpeedLimitType {
    UNKNOWN = 0;
    UNLIMITED = 1;
  }
   oneof speed_limit_oneof {
     uint32 numeric_speed_limit;
     SpeedLimitType other_speed_limit;
   }
   ...
}
```

This would change code tests for `edge.speed_limit() == 255` for `edge.has_numeric_speed_limit()` and `edge.has_other_speed_limit() && edge.other_speed_limit() == SpeedLimitType::UNLIMITED`, which is a bit more wordy, but possibly clearer intent in the code.

Just a suggestion if you have the bandwidth to refactor, not a blocker.